independent-projects/arc/runtime/src/main/java/io/quarkus/arc/InstanceBean.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.arc;
+
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import javax.enterprise.context.spi.CreationalContext;
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.spi.InjectionPoint;
+import javax.enterprise.util.TypeLiteral;
+
+public class InstanceBean extends BuiltInBean<Instance<?>> {
+
+    public static final Set<Type> INSTANCE_TYPES = Collections
+            .unmodifiableSet(new HashSet<>(Arrays.asList(Instance.class, new TypeLiteral<Instance<Object>>() {
+            }.getType(), Object.class)));
+
+    @Override
+    public Set<Type> getTypes() {
+        return INSTANCE_TYPES;
+    }
+
+    @Override
+    public Class<?> getBeanClass() {
+        return InstanceImpl.class;
+    }
+
+    @Override
+    public Instance<?> get(CreationalContext<Instance<?>> creationalContext) {
+        // Obtain current IP to get the required type and qualifiers
+        InjectionPoint ip = InjectionPointProvider.get();
+        return new InstanceImpl<Instance<?>>((InjectableBean<?>) ip.getBean(), ip.getType(),
```

Comment:
I think we do need to add this instance to the parent CC in this case because all `@Dependent` bean instances produced by `Instance` belong to the CC of the bean that injects the `Instance` instance.

Reply:
Oops, you are right, I didn't mean to remove that one.
BTW is it safe to assume that `InjectionPoint.getBean()` always returns instance of `InjectableBean<Instance<?>>`? Because we need to cast it there anyway...

Reply:
It should be safe...